test/new_relic/license_test.rb

```diff
@@ -1,130 +0,0 @@
-# our license data up to date.
-class LicenseTest < Minitest::Test
-  include NewRelic::TestHelpers::FileSearching
-
-  # A list of regexs that will likely match license info
-  LICENSE_TERMS = {
-    'GPL' => /GPL/i,
-    '(c)' => /\(c\)/i,
-    'Copyright' => /copyright/i,
-    'BSD' => /\bBSD\b/i,
-    'MIT' => /\bMIT\b/i,
-    'Apache' => /\bapache\b/i,
-    'rights reserved' => /rights reserved/i
-  }
-
-  # Known occurrences of the above license terms
-  # format is:
-  # [ file, term ] => expected_number_of_occurances
-  # unless listed here the expectation is that these terms will not occur in
-  # the source code.
-  EXPECTED_LICENSE_OCCURRENCES = {
-    ['/LICENSE', '(c)'] => 1, # this one is for item c, not a copyright symbol
-    ['/LICENSE', 'Copyright'] => 12,
-    ['/LICENSE', 'Apache'] => 6,
-    ['/README.md', 'Apache'] => 1,
-    ['/THIRD_PARTY_NOTICES.md', 'Copyright'] => 1,
-    ['/newrelic_rpm.gemspec', 'Apache'] => 1,
-    ['/infinite_tracing/LICENSE', '(c)'] => 1,
-    ['/infinite_tracing/LICENSE', 'Copyright'] => 12,
-    ['/infinite_tracing/LICENSE', 'Apache'] => 6,
-    ['/infinite_tracing/newrelic-infinite_tracing.gemspec', 'Apache'] => 1,
-    ['/CHANGELOG.md', 'BSD'] => 3, # reference to BSD the operating system, not BSD the license
-    ['/lib/new_relic/agent/new_relic_service.rb', 'Apache'] => 1, # reference to Apache Tomcat in comments
-    ['/lib/new_relic/agent/system_info.rb', 'BSD'] => 4, # reference to the operating system, not the license
-    ['/test/new_relic/agent/system_info_test.rb', 'BSD'] => 2 # reference to the operating system, not the license
-  }
-
-  def shebang
-    /^#!/
-  end
-
-  def encoding
-    /^# ?(en)?coding: utf-8/
-  end
-
-  def syntax_mark
-    /^# -\*- ruby -\*-/
-  end
-
-  def should_skip?(path)
-    (
-      # skip directories
-      !File.file?(path) ||
-      # skip binary files and js files
-      %w[ .sqlite3 .log .png .ico .gif .pdf .gem .js ].include?(File.extname(path)) ||
-      # skip this file
-      File.expand_path(__FILE__) == path ||
-      # skip tmp directories overall
-      path.include?('/tmp/') ||
-      # skip vendor/bundle
-      path.include?(gem_root + '/vendor/bundle') ||
-      # skip the auto-generated build.rb file
-      path =~ %r{lib/new_relic/build\.rb} ||
-      # skip the YARD-generated doc directory
-      path.include?(gem_root + '/doc') ||
-      # skip tags file
-      path =~ %r{/tags$}i ||
-      # skip multiverse auto-generated gemfiles
-      path =~ %r{/test/multiverse/suites/.*/Gemfile\.\d+(\.lock)?$} ||
-      # skip multiverse auto-generated db/schema
-      path =~ %r{/test/multiverse/suites/.*/db/schema.rb$} ||
-      # skip multiverse error log output
-      path =~ %r{/test/multiverse/lib/multiverse/errors.txt$} ||
-      # skip the artifacts directory
-      path =~ %r{/artifacts/} ||
-      # skip the changelog
-      path =~ %r{CHANGELOG.md}
-    )
-  end
-
-  def test_all_rb_and_js_files_have_license_header
```

Comment:
I remember talking about the other test (test_for_scary_license_terms), but I don't remember this one being mentioned. Since this is checking to make sure the license header is on all files, is that something we should be checking or is similarly unnecessary as the other test?